client/driver/logging/rotator.go

```diff
@@ -0,0 +1,218 @@
+package logging
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+var (
+	bufSize = 32
+	buf     = make([]byte, bufSize)
+)
+
+// FileRotator writes bytes to a rotated set of files
+type FileRotator struct {
+	// MaxFiles is the maximum number of rotated files allowed in a path
+	MaxFiles int
+	// FileSize is the size a rotated file is allowed to grow
```

Comment:
@dadgar I think the comments should go on the side, since users of the rotator are going to create the FileRotator via the `NewFileRotator` API